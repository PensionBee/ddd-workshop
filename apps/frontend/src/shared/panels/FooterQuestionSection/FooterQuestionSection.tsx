import { GridContainer, GridItem, GridRow } from "@shared/components/Grid";
import { Heading2El, BodySmallL } from "@shared/components/Typography";
import { Link } from "@shared/components/Link";

export const FooterQuestionSection: React.FC = (props) => (
  <GridContainer
    data-rc="FooterQuestionSection"
    backgroundColor="grey5"
    pt={{ xs: 36, md: 48, lg: 56 }}
    pb={{ xs: 36, md: 48, lg: 56 }}
    {...props}
  >
    <GridRow pb={10}>
      <GridItem>
        <Heading2El align="center">
          Have a question? Call our UK team{" "}
          <Link href="tel:+442034578444" nowrap>
            020 3457 8444
          </Link>
        </Heading2El>
      </GridItem>
    </GridRow>
    <GridRow>
      <GridItem>
        <BodySmallL align="center">
          <BodySmallL nowrap>Monday-Wednesday</BodySmallL>{" "}
          <BodySmallL nowrap>9:30am-6pm,</BodySmallL>{" "}
          <BodySmallL nowrap>Thursday-Friday</BodySmallL>{" "}
          <BodySmallL nowrap>9:30am-5pm</BodySmallL>
        </BodySmallL>
      </GridItem>
    </GridRow>
  </GridContainer>
);
