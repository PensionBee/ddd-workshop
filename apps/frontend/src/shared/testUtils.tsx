import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import type { Breakpoints } from "./styling/breakpoints";
import { BREAKPOINTS } from "./styling/breakpoints";
import { theme } from "./styling/theme";
import { vi } from "vitest"

type ReactTestingLibraryProviderProps = {
  children: React.ReactNode;
};

// eslint-disable-next-line react-refresh/only-export-components
const ReactTestingLibraryProviders: React.FC<
  ReactTestingLibraryProviderProps
> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const customRender = (ui: React.ReactElement, options?: any) => {
  return render(ui, { wrapper: ReactTestingLibraryProviders, ...options });
};

// re-export everything
// eslint-disable-next-line react-refresh/only-export-components
export * from "@testing-library/react";

// override render method
export { customRender as render };

export const sleep = (delay: number) => {
  return new Promise((resolve) => setTimeout(resolve, delay));
};

export const mockMediaMatches = (breakpoint: Breakpoints) => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: (query: string) => {
      const breakpointValue = BREAKPOINTS[breakpoint];
      const queryValue = parseInt(query.replace(/^\D+/g, ""));
      const breakpointValues = [...Object.values(BREAKPOINTS)];
      if (!breakpointValues.includes(queryValue)) {
        throw new Error(
          `Invalid breakpoint value: ${queryValue}, this likely means there has been a change to the way that breakpoints are setup.`
        );
      }
      return {
        matches: queryValue <= breakpointValue,
        addEventListener: vi.fn(),
        removeEventListener: vi.fn(),
      };
    },
  });
};

export const routerRender = (ui: React.ReactNode, pathname = "/") => {
  return customRender(
    <MemoryRouter initialEntries={[{ pathname }]}>{ui}</MemoryRouter>
  );
};
