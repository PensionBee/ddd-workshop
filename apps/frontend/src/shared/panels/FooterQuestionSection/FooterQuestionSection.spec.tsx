import { render, vi } from "@shared/testUtils";
import { FooterQuestionSection } from "./FooterQuestionSection";

describe("FooterQuestionSection", () => {
  const TEST_ID = "FooterQuestionSection";
  it("should render", () => {
    const { getByTestId, asFragment } = render(
      <FooterQuestionSection data-testid={TEST_ID} />
    );
    expect(getByTestId(TEST_ID)).toBeInTheDocument();
    expect(asFragment).toMatchSnapshot();
  });
});
