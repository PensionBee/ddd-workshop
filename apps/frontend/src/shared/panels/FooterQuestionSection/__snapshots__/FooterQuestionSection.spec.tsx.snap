// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FooterQuestionSection > should render 1`] = `[Function]`;

exports[`FooterQuestionSection should render 1`] = `[Function]`;
